test_simt.py

```diff
@@ -0,0 +1,17 @@
+import taichi as ti
+
+ti.init(arch=ti.cuda)
+
+a = ti.field(dtype=ti.i32, shape=128)
+
+
+@ti.kernel
+def foo():
+    ti.loop_config(block_dim=32)
+    for i in range(128):
+        a[i] = ti.lang.shfl_down_sync_i32(0x7FFFFFFF, i * 100, 2)
```

Comment:
I cannot use `0x7FFFFFFF` as mask, otherwise Taichi complains
```
        a[i] = ti.lang.shfl_down_sync_i32(0xFFFFFFFF, i * 100, 2)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Constant 4294967295 has exceeded the range of i32: [-2147483648, 2147483647]
```

Probably because LLVM doesn't distinguish signed/unsigned integers. Any idea/easy solution? :-)

Reply:
This is a frontend thing :-) Because the default int type (which the literal adopts) is i32, Taichi checks whether the literal is in the range of i32. You can use `ti.u32(0xFFFFFFFF)`, which is equivalent to `0xFFFFFFFFu` in C.